src/ucp/core/ucp_worker.c

```diff
@@ -2911,41 +2915,48 @@ ucp_worker_discard_tl_uct_ep(ucp_ep_h ucp_ep, uct_ep_h uct_ep,
     req->send.state.uct_comp.status         = UCS_OK;
     req->send.discard_uct_ep.uct_ep         = uct_ep;
     req->send.discard_uct_ep.ep_flush_flags = ep_flush_flags;
+    ucp_request_set_callback(req, send.cb, discarded_cb, discarded_cb_arg);
 
     ucp_worker_discard_uct_ep_progress(req);
 }
 
 static void
 ucp_worker_discard_wireup_uct_ep(ucp_ep_h ucp_ep, ucp_wireup_ep_t *wireup_ep,
-                                 unsigned ep_flush_flags, uct_ep_h uct_ep)
+                                 unsigned ep_flush_flags, uct_ep_h uct_ep,
+                                 ucp_send_nbx_callback_t discarded_cb,
+                                 void *discarded_cb_arg)
 {
     if (uct_ep == NULL) {
         return;
     }
 
     ucp_wireup_ep_disown(&wireup_ep->super.super, uct_ep);
-    /* discard the WIREUP EP's UCT EP */
-    ucp_worker_discard_uct_ep(ucp_ep, uct_ep, ep_flush_flags,
-                              /* make sure that there are no WIREUP MSGs
-                               * anymore that are scheduled on the UCT EP, i.e.
-                               * the purge callback hasn't be invoked here */
-                              (uct_pending_purge_callback_t)
-                              ucs_empty_function_do_assert, NULL);
+    /* Discard the WIREUP EP's UCT EP */
+    ucp_worker_discard_uct_ep(
+            ucp_ep, uct_ep, ep_flush_flags,
+            /* Make sure that there are no WIREUP MSGs anymore that are
+               scheduled on the UCT EP, i.e. the purge callback hasn't be
+               invoked here */
+            (uct_pending_purge_callback_t)ucs_empty_function_do_assert, NULL,
+            discarded_cb, discarded_cb_arg);
 }
 
-static uct_ep_h
-ucp_worker_discard_wireup_ep(ucp_ep_h ucp_ep, ucp_wireup_ep_t *wireup_ep,
-                             unsigned ep_flush_flags,
-                             uct_pending_purge_callback_t purge_cb,
-                             void *purge_arg)
+static uct_ep_h ucp_worker_discard_wireup_ep(
+        ucp_ep_h ucp_ep, ucp_wireup_ep_t *wireup_ep, unsigned ep_flush_flags,
+        uct_pending_purge_callback_t purge_cb, void *purge_arg,
+        ucp_send_nbx_callback_t discarded_cb, void *discarded_cb_arg)
 {
+    ucp_worker_h worker = ucp_ep->worker;
+    ucp_worker_iface_t *wiface;
     uct_ep_h uct_ep;
     int is_owner;
 
     ucs_assert(wireup_ep != NULL);
 
+    wiface = ucp_worker_iface(worker, wireup_ep->aux_rsc_index);
     ucp_worker_discard_wireup_uct_ep(ucp_ep, wireup_ep, ep_flush_flags,
-                                     wireup_ep->aux_ep);
+                                     wireup_ep->aux_ep,
+                                     ucp_wireup_ep_aux_ep_discarded, wiface);
```

Comment:
shall we add common helper function ucp_worker_discard_aux_uct_ep ?

Reply:
fixed